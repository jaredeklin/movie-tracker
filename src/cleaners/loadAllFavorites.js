const url = process.env.REACT_APP_DATABASE_URL;

export const loadAllFavorites = async (userId) => {

  try {
    const response = await fetch(`${url}/api/v1/users/${userId}/favorites`);
    if (response.status === 404) {
      return {};
    }
    const favoriteData = await response.json();
    
    return favoriteData;
  } catch (error) {
    throw error;
  }
};