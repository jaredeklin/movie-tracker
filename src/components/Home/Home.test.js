import React from 'react';
import { Home } from './Home';
import { mockCleanNowPlaying, mockFavorites } from '../../cleaners/mockData';
import { shallow } from 'enzyme';


describe('Home', () => {

  it('should match the snapshot', () => {

    let wrapper = shallow(<Home 
      movies={mockCleanNowPlaying} 
      favorites={mockFavorites} />);

    expect(wrapper).toMatchSnapshot();
  });

});