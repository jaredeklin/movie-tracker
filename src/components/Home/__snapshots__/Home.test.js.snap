// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    favorites={
        Array [
            Object {
              "movie_id": "284054",
              "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
              "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
              "release_date": "2018-02-13",
              "title": "Black Panther",
              "vote_average": "7.4",
            },
          ]
    }
    movies={
        Array [
            Object {
              "movie_id": "284054",
              "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
              "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
              "release_date": "2018-02-13",
              "title": "Black Panther",
              "vote_average": "7.4",
            },
            Object {
              "movie_id": "284000",
              "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
              "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
              "release_date": "2018-02-13",
              "title": "Black Panther",
              "vote_average": "7.4",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="inner-card-container"
>
        <Connect(Card)
                movieInfo={
                        Object {
                                "movie_id": "284054",
                                "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
                                "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                                "release_date": "2018-02-13",
                                "title": "Black Panther",
                                "vote_average": "7.4",
                              }
                }
                selected="selected"
        />
        <Connect(Card)
                movieInfo={
                        Object {
                                "movie_id": "284000",
                                "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
                                "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                                "release_date": "2018-02-13",
                                "title": "Black Panther",
                                "vote_average": "7.4",
                              }
                }
                selected=""
        />
</div>,
      "className": "card-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Card)
            movieInfo={
                        Object {
                                    "movie_id": "284054",
                                    "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
                                    "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                                    "release_date": "2018-02-13",
                                    "title": "Black Panther",
                                    "vote_average": "7.4",
                                  }
            }
            selected="selected"
/>,
          <Connect(Card)
            movieInfo={
                        Object {
                                    "movie_id": "284000",
                                    "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
                                    "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                                    "release_date": "2018-02-13",
                                    "title": "Black Panther",
                                    "vote_average": "7.4",
                                  }
            }
            selected=""
/>,
        ],
        "className": "inner-card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "movieInfo": Object {
              "movie_id": "284054",
              "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
              "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
              "release_date": "2018-02-13",
              "title": "Black Panther",
              "vote_average": "7.4",
            },
            "selected": "selected",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "movieInfo": Object {
              "movie_id": "284000",
              "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
              "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
              "release_date": "2018-02-13",
              "title": "Black Panther",
              "vote_average": "7.4",
            },
            "selected": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inner-card-container"
>
          <Connect(Card)
                    movieInfo={
                              Object {
                                        "movie_id": "284054",
                                        "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
                                        "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                                        "release_date": "2018-02-13",
                                        "title": "Black Panther",
                                        "vote_average": "7.4",
                                      }
                    }
                    selected="selected"
          />
          <Connect(Card)
                    movieInfo={
                              Object {
                                        "movie_id": "284000",
                                        "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
                                        "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                                        "release_date": "2018-02-13",
                                        "title": "Black Panther",
                                        "vote_average": "7.4",
                                      }
                    }
                    selected=""
          />
</div>,
        "className": "card-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Card)
              movieInfo={
                            Object {
                                          "movie_id": "284054",
                                          "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
                                          "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                                          "release_date": "2018-02-13",
                                          "title": "Black Panther",
                                          "vote_average": "7.4",
                                        }
              }
              selected="selected"
/>,
            <Connect(Card)
              movieInfo={
                            Object {
                                          "movie_id": "284000",
                                          "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
                                          "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                                          "release_date": "2018-02-13",
                                          "title": "Black Panther",
                                          "vote_average": "7.4",
                                        }
              }
              selected=""
/>,
          ],
          "className": "inner-card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "movieInfo": Object {
                "movie_id": "284054",
                "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
                "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                "release_date": "2018-02-13",
                "title": "Black Panther",
                "vote_average": "7.4",
              },
              "selected": "selected",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "movieInfo": Object {
                "movie_id": "284000",
                "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
                "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
                "release_date": "2018-02-13",
                "title": "Black Panther",
                "vote_average": "7.4",
              },
              "selected": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
